packages/alfa-aria/src/node.ts

```diff
@@ -222,7 +223,65 @@ export namespace Node {
 
     const root = node.root({ flattened: true });
 
-    build(root, device);
+    // Before we start constructing the accessibility tree, we need to resolve
+    // explicit ownership of elements as specified by the `aria-owns` attribute.
+    // https://w3c.github.io/aria/#aria-owns
+
+    // Find all elements in the tree. As explicit ownership is specified via ID
+    // references, it cannot cross shadow or document boundaries.
+    const elements = root.inclusiveDescendants().filter(dom.Element.isElement);
+
+    // Build a map from ID -> element to allow fast resolution of ID references.
+    // The collected references are added to the map in reverse order to ensure
+    // that the first occurrence of a given ID is what ends up in the map in
+    // event of duplicates.
+    const ids = Map.from(
+      elements
+        .collect((element) => element.id.map((id) => [id, element] as const))
+        .reverse()
+    );
+
+    // Do a first pass over `aria-owns` attributes and collect the referenced
+    // elements.
+    const references = elements.collect((element) =>
+      element.attribute("aria-owns").map(
+        (attribute) =>
+          [
+            element,
+            attribute
+              .tokens()
+              .collect((id) => ids.get(id))
+              // Reject select-references.
+              .reject((reference) => reference === element),
+          ] as const
+      )
+    );
+
+    // Refine the collected `aria-owns` references, constructing a set of
+    // claimed elements and resolving conflicting claims as needed.
+    const [claimed, owned] = references.reduce(
+      ([claimed, owned], [element, references]) => {
+        // Reject all element references that have already been claimed.
```

Comment:
As far as I've understood, specs don't say how to resolve multiple claim and we are making a choice to give them to the first claimer.
I guess we should document that this is our choice and a spec ambiguity so that it's easier to find and fix if specs are fixed…

Reply:
This is specifically left as undefined behaviour:

> Authors **MUST** ensure that an element's ID is not specified in more than one other element's `aria-owns` attribute at any time. In other words, an element can have only one explicit owner.

I'll add an explanatory comment 👍